platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Link.kt

```diff
@@ -149,6 +161,110 @@ public fun ExternalLink(
     )
 }
 
+/**
+ * An external link that follows the standard visual styling, including an external link icon.
+ *
+ * Please be aware that this ExternalLink will automatically open a link on click, unlike the other.
+ *
+ * Provides a text link with an external link icon that indicates the link leads to external content. The link supports
+ * various states including enabled/disabled, focused, and hovered, with optional underline behavior based on the style
+ * configuration.
+ *
+ * **Guidelines:** [on IJP SDK webhelp](https://plugins.jetbrains.com/docs/intellij/link.html#external-link-icon)
+ *
+ * **Usage example:**
+ * [`Links.kt`](https://github.com/JetBrains/intellij-community/blob/master/platform/jewel/samples/standalone/src/main/kotlin/org/jetbrains/jewel/samples/standalone/view/component/Links.kt)
+ *
+ * **Swing equivalent:** [`JLabel`](https://docs.oracle.com/javase/tutorial/uiswing/components/label.html) with HTML
+ * link styling and external link icon
+ *
+ * @param text The text to be displayed as a link
+ * @param link The actual link that will be used to open in the browser. Defaults to the text parameter value.
+ * @param onClick Called after the link was opened. Returns whether the opening was successful or not.
+ * @param beforeOpeningLink Called before trying to open the link.
+ * @param modifier Modifier to be applied to the link
+ * @param enabled Controls whether the link can be interacted with
+ * @param textStyle The typography style to be applied to the link text
+ * @param overflow How the text should handle overflow
+ * @param interactionSource Source of interactions for this link
+ * @param style The visual styling configuration for the link
+ * @see javax.swing.JLabel
+ */
+@Composable
+public fun ExternalLink(
+    text: String,
+    link: String = text,
+    onClick: (linkOpened: Boolean) -> Unit,
+    beforeOpeningLink: () -> Unit = {},
```

Comment:
I don't think we should have these. If people care about what happens when the link is clicked, they can use the other overloads, which give them control. This is supposed to be the "easy, works out of the box" version that covers the majority of simple cases.